os-ui/src/redux/Slices/UpdateQuestionSlice.tsx

```diff
@@ -0,0 +1,167 @@
+import {
+  createSlice, createAsyncThunk, PayloadAction,
+} from '@reduxjs/toolkit';
+import { Question } from '../../pages/CourseModulePage/DiscussionPage/interfaces/interfaces';
+import fetchData from '../../services/fetchData';
+import {
+  EditQuestionPayload, PeersState, PostQuestionsPayload, RemoveQuestionPayload,
+} from '../interfaces/UpdateQuestionSliceInterface';
+
+// Define the initial state for the "peers" slice
+const initialState: PeersState = {
+  questionsWithId: [],
+  questionsWithIdToMentor: [],
+  isLoading: false,
+  isOpen: false,
+  section: true,
+  errorMessage: '',
+};
+
+// Define an async thunk for posting questions to the server
+export const addQuestion = createAsyncThunk(
+  'discussionForum/addQuestion',
+  async (payload: PostQuestionsPayload, { rejectWithValue }) => {
+    try {
+      const {
+        enrolledCourseId, text, token, sectionName,
+      } = payload;
+      const body = {
+        text,
+      };
+      const response = await fetchData.post(
+        `courses/enrolled/${enrolledCourseId}/${sectionName}-questions`,
+        body,
+        {},
+        token,
+      );
+      const responseJSON = await response.json();
+      const { id, createdDate } = responseJSON;
+      const newQuestion: Question = {
+        id,
+        text,
+        createdDate,
+      };
+      return newQuestion;
+    } catch (error) {
+      return rejectWithValue('Failed to fetch peers questions');
+    }
+  },
+);
```

Comment:
Since we have async functions in CourseService for posting, updating, and deleting peer questions, there's no need to import fetchData. Instead, you can use CourseService.postPeerQuestion, CourseService.putPeerQuestion, and CourseService.deletePeerQuestion. Also, for each request, you should create a separate slice with specific logic. For example, you could create PostPeerQuestionSlice.tsx, PutPeerQuestionSlice.tsx, and DeletePeerQuestionSlice.tsx

Reply:
I have an single array of questions, so I think it is better and easier to have all the logic in the one slice,  becouse I don't want define initialState many times and and make it difficult and dublicate code every time. And CourseService.postPeerQuestion, this will post question only for peers, instead my code ca post both for peers and mentors.